app/src/androidTest/java/com/team7/cmput301/android/theirisproject/EditProblemActivityTest.java

```diff
@@ -0,0 +1,152 @@
+
+import com.robotium.solo.Solo;
+import com.team7.cmput301.android.theirisproject.activity.EditProblemActivity;
+
+import com.team7.cmput301.android.theirisproject.activity.ViewProblemActivity;
+import com.team7.cmput301.android.theirisproject.helper.Timer;
+import com.team7.cmput301.android.theirisproject.model.Patient;
+import com.team7.cmput301.android.theirisproject.model.Problem;
+import com.team7.cmput301.android.theirisproject.task.AddProblemTask;
+import com.team7.cmput301.android.theirisproject.task.Callback;
+
+import java.text.ParseException;
+
+/**
+ * Test for EditProblemActivity
+ *
+ * @author VinnyLuu
+ */
+public class EditProblemActivityTest extends ActivityInstrumentationTestCase2<EditProblemActivity> {
+
+    private Solo solo;
+    private String userName = "John Doe";
+    private String userEmail = "johndoe@gmail.com";
+    private String userPhone = "123-456-7890";
+    private String patientID = "111";
+    private String probTitle = "ouch";
+    private String probDesc = "I'm shot";
+    private String probDate = "2018-09-01T18:30:00";
+    private String probID = "123";
+
+    public EditProblemActivityTest() {
+        super(EditProblemActivity.class);
+    }
+
+    @Override
+    protected void setUp() {
+        Patient patient = new Patient(userName, userEmail, userPhone);
+        patient.setId(patientID);
+        IrisProjectApplication.setCurrentUser(patient);
+        Problem problem = null;
+        try {
+            problem = new Problem(probTitle, probDesc, probDate , patientID);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        problem.setId(probID);
+
+
+        new AddProblemTask(new Callback<String>()  {
+            @Override
+            public void onComplete(String result) {
+                // Don not need result
+            }
+        }).execute(problem);
+        Intent intent = new Intent();
+        intent.putExtra(ViewProblemActivity.EXTRA_PROBLEM_ID, problem.getId());
+        setActivityIntent(intent);
+        solo = new Solo(getInstrumentation(), getActivity());
+    }
+
+    @Override
+    protected void tearDown() {
+        solo.finishOpenedActivities();
+    }
+
+    public void testActivity() {
+        String activityName = EditProblemActivity.class.getSimpleName();
+        solo.assertCurrentActivity("Wrong activity found, should be " + activityName, activityName);
+    }
+
+    /**
+     * Test if problem details are displayed correctly
+     */
+    public void testCorrectProblemData() {
+
+        Timer.sleep(1000);
+        EditText title = (EditText) solo.getView(R.id.problem_title);
+        EditText desc = (EditText) solo.getView(R.id.problem_description);
+        EditText date = (EditText) solo.getView(R.id.problem_date);
+
+        assertTrue(title.getText().toString().equals(probTitle));
+        assertTrue(desc.getText().toString().equals(probDesc));
+        assertTrue(date.getText().toString().equals(probDate));
+    }
```

Comment:
Should also click the Button to finish editing, and then waitForActivity to check that it properly goes to where it's supposed to